framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -7556,22 +7556,63 @@ void VulkanReplayConsumerBase::OverrideCmdBuildAccelerationStructuresKHR(
-    // Use the builder when the rebind allocator is selected and the trimming is done / not used
-    else if (!loading_trim_state_)
+    DeviceInfo*     device_info    = object_info_table_.GetDeviceInfo(command_buffer_info->parent_id);
+    VkCommandBuffer command_buffer = command_buffer_info->handle;
+    VkAccelerationStructureBuildGeometryInfoKHR* build_geometry_infos = pInfos->GetPointer();
+    VkAccelerationStructureBuildRangeInfoKHR**   build_range_infos    = ppBuildRangeInfos->GetPointer();
+
+    auto& address_tracker = GetDeviceAddressTracker(device_info->handle);
+
+    auto address_remap = [&address_tracker](const VkDeviceAddress& capture_address) {
+        auto buffer_info = address_tracker.GetBufferByCaptureDeviceAddress(capture_address);
+        GFXRECON_ASSERT(buffer_info != nullptr);
+
+        uint64_t offset = capture_address - buffer_info->capture_address;
+
+        // in-place address-remap via const-cast
+        auto& dst = *const_cast<VkDeviceAddress*>(&capture_address);
+        dst       = buffer_info->replay_address + offset;
+    };
+
+    for (uint32_t i = 0; i < infoCount; ++i)
     {
-        GFXRECON_ASSERT(false);
-        //        acceleration_structure_builders_[command_buffer_info->parent_id]->CmdBuildAccelerationStructures(
-        //            command_buffer_info->handle, infoCount, pInfos->GetPointer(), ppBuildRangeInfos->GetPointer());
+        const auto& build_geometry_info = build_geometry_infos[i];
+
+        for (uint32_t j = 0; j < build_geometry_info.geometryCount; ++j)
+        {
+            const VkAccelerationStructureGeometryKHR* geometry = build_geometry_info.pGeometries != nullptr
+                                                                     ? build_geometry_info.pGeometries + j
+                                                                     : build_geometry_info.ppGeometries[j];
+
+            switch (geometry->geometryType)
+            {
+                case VK_GEOMETRY_TYPE_TRIANGLES_KHR:
+                {
+                    const auto& triangles = geometry->geometry.triangles;
+                    address_remap(triangles.vertexData.deviceAddress);
+                    address_remap(triangles.indexData.deviceAddress);
+                    break;
+                }
+                case VK_GEOMETRY_TYPE_AABBS_KHR:
+                {
+                    const auto& aabbs = geometry->geometry.aabbs;
+                    address_remap(aabbs.data.deviceAddress);
+                    break;
+                }
+                case VK_GEOMETRY_TYPE_INSTANCES_KHR:
+                {
+                    const auto& instances = geometry->geometry.instances;
+                    // TODO: inside buffer:
+                    // replace all VkAccelerationStructureInstanceKHR::accelerationStructureReference
+                    address_remap(instances.data.deviceAddress);
+                    break;
+                }
+                default:
+                    break;
```

Comment:
Please add an error and/or assert here so we can catch missing cases in the future.

Reply:
done